swerve_drive/subsystems/drive.py

```diff
@@ -45,15 +47,39 @@ def __init__(self) -> None:
         self.steer_3 = TalonFX(TalonIds.steer_2)
         self.steer_4 = TalonFX(TalonIds.steer_4)
 
+        self.encoder_1 = CANcoder(CancoderIds.swerve_1)
+        self.encoder_2 = CANcoder(CancoderIds.swerve_2)
+        self.encoder_3 = CANcoder(CancoderIds.swerve_3)
+        self.encoder_4 = CANcoder(CancoderIds.swerve_4)
+
         self.steer_motors = [self.steer_1, self.steer_2, self.steer_3, self.steer_4]
-        for steer_motor in self.steer_motors:
+        self.steer_encoders = [
+            self.encoder_1,
+            self.encoder_2,
+            self.encoder_3,
+            self.encoder_4,
+        ]
+        for steer_motor, steer_encoder in zip(self.steer_motors, self.steer_encoders):
             steer_motor_config = MotorOutputConfigs()
             steer_motor_config.neutral_mode = NeutralModeValue.BRAKE
+            steer_pid = Slot0Configs().with_k_p(211.73).with_k_i(0).with_k_d(27.368)
```

Comment:
Using the PID gains we got today:
```suggestion
            steer_pid = Slot0Configs().with_k_p(4.1615).with_k_i(0).with_k_d(0.0021683)
```